transport/socks5/socks5.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package socks5
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+)
+
+const addrTypeIPv4 = 0x01
+const addrTypeDomainName = 0x03
+const addrTypeIPv6 = 0x04
+
+func appendSOCKS5Address(b []byte, address string) ([]byte, error) {
```

Comment:
*nit* maybe it's worth to put a reference link or just a bytes diagram of the related SOCK5 protocol to this function?

Reply:
Good idea. Added.